pages/components/FlowSim.tsx

```diff
@@ -0,0 +1,116 @@
+import { ChangeEvent, KeyboardEvent, useEffect, useState } from "react";
+import { Member } from "@/lib/models";
+
+interface Props {
+    flowId: number
+}
+
+type Message = string // likely to evolve pretty quickly
+
+async function simulateFlow(
+    flowId: number,
+    member: Member,
+    message: Message = '',
+    index: number = 0
+) {
+    const res = await fetch(`/api/flows/${flowId}/simulate`, {
+        method: 'POST',
+        headers: {
+            'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({
+            member,
+            message,
+            startIndex: index,
+        }),
+    })
+
+    if (!res.ok) {
+        throw new Error(`HTTP Error. Status: ${res.status}`)
+    }
+
+    return res.json()
+}
+
+export default function FlowSim(props: Props) {
+    // This format provided for convenience. Please change if necessary.
+    const [messages, setMessages] = useState<Message[]>([]);
+    const [submission, setSubmission] = useState<Message>('');
+    const [input, setInput] = useState<string>('');
+    const [index, setIndex] = useState(0);
+
+    // TODO: Switch to real memeber when needed.
+    const member: Member = {
+        id: 123,
+        name: 'fake-name',
+        phoneNumber: 'fake-phone-number',
+        email: 'fake-email',
+        isSubscribed: true,
+    }
+
+    useEffect(() => {
+        simulateFlow(props.flowId, member, "", 0)
+            .then((res) => {
+                if (res.ok) {
+                    setMessages(res.messages);
+                    setIndex(res.stopIndex)
+                    setInput("");
+                } else {
+                    console.error("SeverError: ", res.error);
+                }
+
+                setInput('');
+            })
+            .catch(console.error)
+    }, [props.flowId]);
+
+    useEffect(() => {
+        if (submission == '') return;
+
+        simulateFlow(props.flowId, member, submission, index)
+            .then((res) => {
+                if (res.ok) {
+                    setMessages(messages.concat(res.messages));
```

Comment:
concating messages on the client would probably be one of the first strategies to go if we were doing this for real. Obviously chats would need to be persisted somewhere, but even if you didn't go that far, it would be better to send the complete conversation back and forth.